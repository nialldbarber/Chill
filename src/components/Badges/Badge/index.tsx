export {default as Badges} from './Badge';
