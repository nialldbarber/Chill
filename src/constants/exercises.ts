export const IN = 'In';
export const OUT = 'Out';
export const HOLD = 'Hold';

const page = 'Exercise';

export const feelings = [
  'activated',
  'alleviated',
  'awake',
  'balanced',
  'blissful',
  'calm',
  'clear',
  'cool',
  'dream',
  'euphoric',
  'energized',
  'focused',
  'pain relief',
  'peaceful',
  'recharged',
  'relaxed',
  'rise and shine',
  'sleepy',
  'stimulated',
  'tranquil',
  'uplifted',
  'unwind',
];

export const FEELINGS_COLOR_MAP: Record<string, string> = {
  calm: 'calm',
  energy: 'primary',
  night: 'night',
};

export type ConfigT = {
  id: number;
  page: string;
  exerciseName: string;
  exercise: number[];
  type?: number;
  theme: string;
  category?: string;
  tags?: string[];
};

// categories
// - calm
// - energy
// - night

// types
// - 1 = in hold out hold
// - 2 = in out hold
// - 3 = in hold out
// - 4 = in out

export const CONFIG: ConfigT[] = [
  {
    id: 1,
    page,
    exerciseName: 'Box',
    exercise: [4, 4, 4, 4],
    type: 1,
    theme: 'dark',
    category: 'calm',
    tags: [
      'alleviated',
      'balanced',
      'calm',
      'clear',
      'cool',
      'peaceful',
      'relaxed',
      'tranquil',
    ],
  },
  {
    id: 2,
    page,
    exerciseName: 'Calm',
    exercise: [4, 7, 8, 0],
    type: 2,
    theme: 'dark',
    category: 'calm',
    tags: ['uplifted'],
  },
  {
    id: 3,
    page,
    exerciseName: 'Deep',
    exercise: [5, 0, 5, 0],
    type: 4,
    theme: 'dark',
    category: 'calm',
    tags: ['uplifted'],
  },
  {
    id: 4,
    page,
    exerciseName: 'Awake',
    exercise: [6, 0, 2, 0],
    type: 4,
    theme: 'dark',
    category: 'energy',
    tags: [
      'activated',
      'awake',
      'euphoric',
      'energized',
      'rise and shine',
      'stimulated',
    ],
  },
  {
    id: 5,
    page,
    exerciseName: 'Recharge',
    exercise: [5, 3, 4, 3],
    type: 1,
    theme: 'dark',
    category: 'energy',
    tags: [
      'activated',
      'awake',
      'energized',
      'focused',
      'recharged',
      'rise and shine',
      'stimulated',
    ],
  },
  // NIGHT
  {
    id: 6,
    page,
    exerciseName: 'Dream',
    exercise: [5, 15, 5, 0],
    type: 3,
    theme: 'dark',
    category: 'night',
    tags: [
      'balanced',
      'calm',
      'dream',
      'peaceful',
      'relaxed',
      'sleepy',
      'tranquil',
      'unwind',
    ],
  },
];
