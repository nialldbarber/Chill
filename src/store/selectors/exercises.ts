import {RootState} from '~/store/index';

export const selectBadges = (state: RootState) => state.exercises.exerciseList;
