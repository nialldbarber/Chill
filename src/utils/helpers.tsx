import React from 'react';

import {Provider} from 'react-redux';

import {store} from '~/store/index';

export function renderWithWrapper(children) {
  return <Provider store={store}>{children}</Provider>;
}
