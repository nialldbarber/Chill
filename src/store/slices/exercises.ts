import {createSlice, PayloadAction} from '@reduxjs/toolkit';
import {CONFIG, ConfigT} from '../../constants/ exercises';

export interface ExerciseState {
  exerciseList: Array<ConfigT>;
  originalList: Array<ConfigT>;
}

const initialState: ExerciseState = {
  exerciseList: CONFIG,
  originalList: CONFIG,
};

export const exerciseSlice = createSlice({
  name: 'exercises',
  initialState,
  reducers: {
    filterBySelectedBadge: (state, action: PayloadAction<string>) => {
      state.exerciseList = state.originalList;
      state.exerciseList = state.exerciseList.filter((item) => {
        return item?.tags?.includes(action.payload);
      });
    },
  },
});

export const {filterBySelectedBadge} = exerciseSlice.actions;

export default exerciseSlice.reducer;
