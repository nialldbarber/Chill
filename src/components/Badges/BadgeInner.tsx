import React from 'react';
import {StyleSheet, Text, TouchableOpacity} from 'react-native';
import {useDispatch} from 'react-redux';
import Animated, {
  useSharedValue,
  useAnimatedStyle,
  withSpring,
} from 'react-native-reanimated';
import {useTheme} from '@react-navigation/native';
import {
  widthPercentageToDP as wp,
  heightPercentageToDP as hp,
} from 'react-native-responsive-screen';
import {feelings} from '~/constants/exercises';
import {filterBySelectedBadge} from '../../store/slices/exercises';

type BadgeInnerProps = {
  i: number;
  item: string;
  press: () => void;
};

export default function BadgeInner({i, item, press}: BadgeInnerProps) {
  const {colors} = useTheme();

  const styles = StyleSheet.create({
    badgeContainer: {
      marginTop: hp('2%'),
      marginBottom: hp('4%'),
    },
    badge: {
      backgroundColor: colors.primaryFaded,
      paddingHorizontal: 12,
      paddingVertical: 6,
      borderRadius: 20,
      marginRight: 10,
      borderColor: colors.primary,
      borderWidth: 1,
    },
    badgeText: {
      color: colors.primary,
      fontWeight: '800',
      fontSize: wp('4%'),
    },
  });

  const dispatch = useDispatch();

  const scale = useSharedValue(1);
  const scaleStyles = useAnimatedStyle(() => ({
    transform: [{scale: scale.value}],
  }));

  function handleFilter(): void {
    setTimeout(() => {
      dispatch(filterBySelectedBadge(feelings[i]));
    }, 200);
  }

  return (
    <TouchableOpacity
      key={i}
      activeOpacity={1}
      onPress={() => {
        handleFilter();
        press();
      }}
      onPressIn={() => (scale.value = withSpring(1.05))}
      onPressOut={() => (scale.value = withSpring(1))}
      style={styles.badgeContainer}
    >
      <Animated.View
        style={[
          styles.badge,
          {marginLeft: i === 0 ? wp('5%') : 0},
          scaleStyles,
        ]}
      >
        <Text style={styles.badgeText}>{item}</Text>
      </Animated.View>
    </TouchableOpacity>
  );
}
