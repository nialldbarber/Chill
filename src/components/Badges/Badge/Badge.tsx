import React from 'react';

import {BadgeInner} from '~/components/Badges/BadgeInner';
import Scroll from '~/components/helpers/Scrollview';

type BadgesProps = {
  data?: any;
  press?: () => void;
};

type T = number;

export default function Badges({data, press}: BadgesProps) {
  return (
    <Scroll h>
      {data.map((item: T, index: T) => (
        <BadgeInner
          key={index}
          index={index}
          item={item}
          press={() => press && press()}
        />
      ))}
    </Scroll>
  );
}
