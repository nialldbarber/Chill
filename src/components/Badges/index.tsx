import React from 'react';

import BadgeInner from '~/components/Badges/BadgeInner';
import Scroll from '~/components/helpers/Scrollview';
import {feelings} from '~/constants/exercises';

export default function Badges({press}: {press: () => void}) {
  return (
    <Scroll h>
      {feelings.map((item, i) => (
        <BadgeInner key={i} {...{i, item, press}} />
      ))}
    </Scroll>
  );
}
