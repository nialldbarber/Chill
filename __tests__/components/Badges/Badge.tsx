import React from 'react';

import renderer from 'react-test-renderer';

import {Badges} from '~/components/Badges/Badge';
import {feelingCategories} from '~/constants/exercises';
import {renderWithWrapper} from '~/utils/helpers';

describe('<Badge />', () => {
  it('should match snapshot', () => {
    const badge = renderer
      .create(renderWithWrapper(<Badges data={feelingCategories} />))
      .toJSON();
    expect(badge).toMatchSnapshot();
  });
});
