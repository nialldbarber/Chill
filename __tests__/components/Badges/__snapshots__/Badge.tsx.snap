// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should match snapshot 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  showsVerticalScrollIndicator={false}
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": undefined,
          "left": undefined,
          "right": undefined,
          "top": undefined,
        }
      }
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 53.5,
          "marginTop": 26.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgba(156, 226, 255, 0.45)",
              "borderColor": "#2098c9",
              "borderRadius": 20,
              "borderWidth": 1,
              "marginRight": 10,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            },
            Object {
              "marginLeft": 37.5,
            },
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#2098c9",
              "fontSize": 30,
              "fontWeight": "800",
            }
          }
        >
          calm
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": undefined,
          "left": undefined,
          "right": undefined,
          "top": undefined,
        }
      }
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 53.5,
          "marginTop": 26.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": undefined,
              "borderColor": undefined,
              "borderRadius": 20,
              "borderWidth": 1,
              "marginRight": 10,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            },
            Object {
              "marginLeft": 0,
            },
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": 30,
              "fontWeight": "800",
            }
          }
        >
          energy
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        Object {
          "bottom": undefined,
          "left": undefined,
          "right": undefined,
          "top": undefined,
        }
      }
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "marginBottom": 53.5,
          "marginTop": 26.5,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": undefined,
              "borderColor": undefined,
              "borderRadius": 20,
              "borderWidth": 1,
              "marginRight": 10,
              "paddingHorizontal": 12,
              "paddingVertical": 6,
            },
            Object {
              "marginLeft": 0,
            },
            Object {
              "transform": Array [
                Object {
                  "scale": 1,
                },
              ],
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": 30,
              "fontWeight": "800",
            }
          }
        >
          night
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
